code/go/0chain.net/smartcontract/stakepool/stakepool.go

```diff
@@ -326,3 +326,20 @@ func (sp *StakePool) equallyDistributeRewards(coins currency.Coin, spUpdate *Sta
 
 	return nil
 }
+
+func (sp *StakePool) SlashFraction(
+	fraction float64,
+	providerId string,
+	providerType spenum.Provider,
+	balances cstate.StateContextI,
+) error {
+	if fraction == 0.0 {
```

Comment:
Perhaps check if the `fraction > 0.0 && fraction <= 1`, otherwise the balance would be negative or 0 after splash.